include/xsparse/util/container_traits.hpp

```diff
@@ -22,7 +100,40 @@ namespace xsparse::util
 
         template <class Key, class Val>
         using Map = TMap<Key, Val>;
+
+        // Check that the container traits have the properly defined methods and method signatures
+        // uses sample type inputs to check the signatures
+        // static_assert(decltype(std::declval<Vec>().contains(std::declval<double>())), std::true_type());
+        // static_assert(std::is_same_v<decltype(std::declval<Vec>().contains(std::declval<double>())), std::true_type>);
+
+        // static_assert(
+        //     has_valid_vec_methods<Vec<double>, double, std::size_t>::value,
+        //     "Vec must have `push_back`, `resize` and `operator[]` methods with the correct signatures.");
+        // static_assert(has_find<Map<int, std::size_t>, std::size_t>::value,
+        //               "Map must have find and resize methods with the correct signatures.");
+        // static_assert(has_contains<Set<std::size_t>, std::size_t>::value,
+        //               "Set must have contain methods with the correct signatures.");
+
+        static_assert(
+            std::is_same_v<decltype(std::declval<Vec<double>>().push_back(std::declval<double>())), void>,
+            "Vec must have `push_back` method with the correct signature.");
+
+        static_assert(
+            std::is_same_v<decltype(std::declval<Vec<double>>().resize(std::declval<typename Vec<double>::size_type>())), void>,
+            "Vec must have `resize` method with the correct signature.");
+
+        static_assert(
+            std::is_same_v<decltype(std::declval<Vec<double>>()[std::declval<std::size_t>()]), decltype(std::declval<Vec<double>>().at(std::declval<std::size_t>()))>,
+            "Vec must have `operator[]` method with the correct signature.");
+
+        static_assert(
+            std::is_same_v<decltype(std::declval<Map<int, std::size_t>>().find(std::declval<std::size_t>())), typename Map<int, std::size_t>::iterator>,
+            "Map must have `find` method with the correct signature.");
+
+        static_assert(
+            std::is_same_v<decltype(std::declval<Set<std::size_t>>().contains(std::declval<std::size_t>())), bool>,
+            "Set must have `contains` method with the correct signature.");
```

Comment:
@hameerabbasi is this what you meant by static assert directly in the container_traits class?

Reply:
Yes. Or maybe in the destructor if that doesn't work.

Reply:
Kay this works and I tried it out by substituting `std::list` for std::vector and as expected, a compiler-error is raised

Reply:
Cool, let's get CI working and then we'll be good to go.

Reply:
Cool @hameerabbasi this is now fixed